pandas/_libs/lib.pyx

```diff
@@ -1472,12 +1472,30 @@ cdef class Validator:
         Py_ssize_t n
         dtype dtype
         bint skipna
+        bint na_only
 
     def __cinit__(self, Py_ssize_t n, dtype dtype=np.dtype(np.object_),
-                  bint skipna=False):
+                  bint skipna=False,
+                  bint na_only=False):
```

Comment:
Alternatively, if people don't like this, I can make a new validator that's specific to StringArray's checking.

```cython
class StringArrayValidator(StringArray):
    cdef bint is_valid_null(self, object value) except -1:
            return value is C_NA
```

Do people have a preference?